fuel-relayer/src/relayer/state/test.rs

```diff
@@ -0,0 +1,89 @@
+use super::{
+    test_builder::TestDataSource,
+    *,
+};
+
+use test_case::test_case;
+
+#[test_case(
+    TestDataSource {
+        eth_remote_current: 300,
+        eth_remote_finalization_period: 100,
+        eth_local_finalized: None,
+    } => Some(0..=200); "empty so needs to sync"
+)]
+#[test_case(
+    TestDataSource {
+        eth_remote_current: 300,
+        eth_remote_finalization_period: 100,
+        eth_local_finalized: Some(0),
+    } => Some(1..=200); "behind so needs to sync"
+)]
+#[test_case(
+    TestDataSource {
+        eth_remote_current: 300,
+        eth_remote_finalization_period: 100,
+        eth_local_finalized: Some(200),
+    } => None; "same so doesn't need to sync"
+)]
+#[test_case(
+    TestDataSource {
+        eth_remote_current: 300,
+        eth_remote_finalization_period: 100,
+        eth_local_finalized: Some(201),
+    } => None; "ahead so doesn't need to sync"
+)]
+#[test_case(
+    TestDataSource {
+        eth_remote_current: 300,
+        eth_remote_finalization_period: 100,
+        eth_local_finalized: Some(50),
+    } => Some(51..=200); "behind by less so needs to sync"
+)]
+#[tokio::test]
```

Comment:
We should probably also test the edge case of what happens when `eth_remote_current < eth_remote_finalization_period`, for new eth nodes (e.g. local dev environment testing)